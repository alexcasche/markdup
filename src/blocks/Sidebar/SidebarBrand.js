import styled from 'styled-components'
import { shade } from 'polished'

const SidebarBrand = styled.h1`
  margin: 0;
  color: ${props => shade(0.8, props.theme.colors.grey)};
  font-size: 28px;
  font-weight: 400;
  letter-spacing: 6px;
  line-height: 86px;
  text-align: center;
  text-transform: uppercase;
  span {
    color: ${props => props.theme.colors.grey};
  }
`

export default SidebarBrand
