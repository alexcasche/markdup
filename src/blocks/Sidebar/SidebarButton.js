import { applyStyleModifiers } from 'styled-components-modifiers'
import { shade } from 'polished'
import { Button } from 'elements'

const MODIFIERS = {
  new: ({ theme }) => `
    margin-top: 15px;
    background-color: ${shade(0.85, theme.colors.blue)};
    &:hover:not(:active) {
      background-color: ${theme.colors.blue};
    }
  `,
  save: ({ theme }) => `
    background-color: ${shade(0.85, theme.colors.green)};
    &:hover:not(:active) {
      background-color: ${theme.colors.green};
    }
  `,
  delete: ({ theme }) => `
    background-color: transparent;
    font-size: 14px;
    border: 1px solid ${shade(0.75, theme.colors.grey)};
    color: ${shade(0.75, theme.colors.grey)};
    &:hover:not(:active) {
      border: 1px solid ${theme.colors.grey};
      color: ${theme.colors.grey};
    }
  `,
}

const SidebarButton = Button.extend`
  margin: 10px auto;
  width: calc(100% - 30px);
  height: 44px;
  font-size: 16px;
  color: ${props => props.theme.colors.black};
  ${applyStyleModifiers(MODIFIERS)};
`

export default SidebarButton
