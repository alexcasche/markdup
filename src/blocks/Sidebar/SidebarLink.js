import { applyStyleModifiers } from 'styled-components-modifiers'
import { tint } from 'polished'
import { Link } from 'elements'

const MODIFIERS = {
  active: ({ theme }) => `
    &, &:hover {
      pointer-events: none;
      cursor: default;
      color: ${theme.colors.grey};
      svg.fa-check-circle {
        color: ${theme.colors.grey};
      }
    }
  `,
}

const SidebarLink = Link.extend`
  position: relative;
  display: block;
  padding: 0 0 0 50px;
  border-bottom: 1px solid ${props => tint(0.85, props.theme.colors.black)};
  color: ${props => props.theme.colors.grey};
  font-size: 14px;
  font-weight: 400;
  line-height: 50px;
  text-align: left;
  text-decoration: none;
  letter-spacing: 1px;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  transition: color 0.15s ease-in-out;
  &:last-child {
    border-bottom: none;
  }
  svg {
    position: absolute;
    right: 20px;
    top: 50%;
    transform: translateY(-50%);
    font-size: 15px;
    transition: inherit;
    &.fa-check-circle {
      color: transparent;
    }
    &.fa-link {
      font-size: 14px;
    }
  }
  &:hover {
    cursor: pointer;
    color: ${props => props.theme.colors.green};
    svg {
      color: ${props => props.theme.colors.green};
    }
    .fa-check-circle {
      color: transparent;
    }
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default SidebarLink
