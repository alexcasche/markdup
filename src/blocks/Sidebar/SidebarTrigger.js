import styled from 'styled-components'
import { tint } from 'polished'

const SidebarTrigger = styled.div`
  display: flex;
  height: 52px;
  align-items: center;
  padding: 0 20px;
  margin-bottom: 2px;
  background-color: ${props => tint(0.85, props.theme.colors.black)};
  color: ${props => props.theme.colors.grey};
  transition: all 0.25s ease-in-out;
  svg:first-child {
    height: 13px;
    width: 20px;
    text-align: center;
    margin-right: 12px;
  }
  span {
    color: inherit;
    flex-grow: 1;
    text-align: left;
    text-transform: uppercase;
    font-size: 14px;
    font-weight: 500;
    letter-spacing: 2px;
  }
  svg:last-child {
    width: 12px;
  }
  &:hover {
    cursor: pointer;
    background-color: ${props => tint(0.8, props.theme.colors.black)};
    color: ${props => props.theme.colors.green};
  }
  &:focus {
    outline: none;
  }
`

export default SidebarTrigger
