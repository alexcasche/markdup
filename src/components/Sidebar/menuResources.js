import { faFileAlt, faLink } from '@fortawesome/free-solid-svg-icons'

const menuResources = () => {
  return {
    title: 'resources',
    icon: faFileAlt,
    links: [
      {
        title: 'Markdown Guide',
        linkProps: { target: '_blank', href: 'https://www.markdownguide.org/' },
      },
      {
        title: 'Github Repository',
        linkProps: {
          target: '_blank',
          href: 'https://github.com/alexcasche/markdup',
        },
      },
    ],
    linkIcon: faLink,
  }
}

export default menuResources
