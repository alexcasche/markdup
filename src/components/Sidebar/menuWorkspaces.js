import { faCode, faCheckCircle } from '@fortawesome/free-solid-svg-icons'

const menuWorkspaces = (allWorkspaces, updateSetting) => {
  let links = []
  Object.keys(allWorkspaces).map(key => {
    const title = allWorkspaces[key].title
    links.push({
      title: title ? title : '--',
      value: key,
      linkProps: {
        onClick: () => updateSetting({ activeWorkspace: key }),
      },
    })
    return links
  })
  return {
    title: 'workspaces',
    icon: faCode,
    links: links,
    linkIcon: faCheckCircle,
  }
}

export default menuWorkspaces
