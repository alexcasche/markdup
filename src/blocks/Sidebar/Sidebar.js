import styled from 'styled-components'
import { tint } from 'polished'

import SidebarBrand from './SidebarBrand'
import SidebarButton from './SidebarButton'
import SidebarTrigger from './SidebarTrigger'
import SidebarLink from './SidebarLink'

const Sidebar = styled.div`
  position: fixed;
  flex-direction: column;
  text-align: center;
  width: 300px;
  height: 100vh;
  font-family: ${props => props.theme.text.fontSans};
  background-color: ${props => tint(0.95, props.theme.colors.black)};
  overflow-y: scroll;
`

Sidebar.Brand = SidebarBrand
Sidebar.Button = SidebarButton
Sidebar.Trigger = SidebarTrigger
Sidebar.Link = SidebarLink

export default Sidebar
