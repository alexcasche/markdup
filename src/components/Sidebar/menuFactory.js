import React from 'react'
import Collapsible from 'react-collapsible'
import { Sidebar } from 'blocks'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faChevronDown } from '@fortawesome/free-solid-svg-icons'
import menuWorkspaces from './menuWorkspaces'
import menuResources from './menuResources'

export const getMenu = (menu, allWorkspaces, updateSetting) => {
  return menu === 'workspaces'
    ? menuWorkspaces(allWorkspaces, updateSetting)
    : menuResources()
}

export const menuTrigger = (title, icon) => {
  return (
    <Sidebar.Trigger>
      <FontAwesomeIcon icon={icon} />
      <span>{title}</span>
      <FontAwesomeIcon icon={faChevronDown} />
    </Sidebar.Trigger>
  )
}

const menuLinks = (links, linkIcon, activeWorkspace) => {
  return links.map((link, index) => {
    const modifiers = [...(activeWorkspace === link.value ? ['active'] : [])]
    return (
      <Sidebar.Link key={index} {...link.linkProps} modifiers={modifiers}>
        {link.title}
        <FontAwesomeIcon icon={linkIcon} />
      </Sidebar.Link>
    )
  })
}

const menuFactory = ({
  menu,
  activeWorkspace,
  allWorkspaces,
  updateSetting,
}) => {
  const menuObj = getMenu(menu, allWorkspaces, updateSetting)
  const { title, icon, links, linkIcon } = menuObj
  return (
    <Collapsible transitionTime={250} trigger={menuTrigger(title, icon)}>
      {menuLinks(links, linkIcon, activeWorkspace)}
    </Collapsible>
  )
}

export default menuFactory
