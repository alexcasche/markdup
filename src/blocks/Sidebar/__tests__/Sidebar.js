import Sidebar from '../Sidebar'

describe('Sidebar Block', () => {
  it('renders correctly', () => {
    const tree = render(<Sidebar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
