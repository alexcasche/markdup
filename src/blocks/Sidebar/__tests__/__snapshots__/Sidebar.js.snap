// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Block renders correctly 1`] = `
.c0 {
  position: fixed;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  width: 300px;
  height: 100vh;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  background-color: #282a2c;
  overflow-y: scroll;
  z-index: 0;
}

<div
  className="c0"
/>
`;
